src/plugins/ldif/ldif.c

```diff
@@ -0,0 +1,420 @@
+	int errnosave = errno;
+	const char * filename = keyString (parentKey);
+	LDIFFP * lfp = ldif_open (filename, "w");
+
+	ELEKTRA_LOG ("Write to '%s'", keyString (parentKey));
+
+	if (!lfp)
+	{
+		ELEKTRA_SET_ERROR_GET (parentKey);
+		errno = errnosave;
+		return ELEKTRA_PLUGIN_STATUS_ERROR;
+	}
+
+	Key * cur;
+	ksRewind (returned);
+	char * last_dn = NULL;
+	while ((cur = ksNext (returned)) != 0)
+	{
+		const char * type = keyBaseName (cur);
+		if (index (type, '=') == NULL)
+		{
+			ELEKTRA_LOG_DEBUG ("%s is a valid ldif type, proceeding writing\n", type);
+			if (elektraStrCmp (type, "dn") != 0)
+			{
+				ELEKTRA_LOG_DEBUG ("%s is not the dn, looking up the corresponding dn first\n", type);
+				const char * full_key_name = elektraStrDup (keyName (cur));
+				size_t full_key_length = elektraStrLen (full_key_name);
+				size_t dn_length = elektraStrLen ("dn");
+				size_t base_name_length = elektraStrLen (type);
+				char * key_dn = elektraMalloc ((full_key_length + dn_length + 1) * sizeof (char));
+				strncpy (key_dn, full_key_name, full_key_length + 1);
+				strncpy (&key_dn[full_key_length - base_name_length], "dn\0", dn_length + 1);
+
+				if (last_dn == NULL || elektraStrCmp (last_dn, key_dn) != 0)
+				{
+					ELEKTRA_LOG_DEBUG ("first occurrence with dn %s, writing it\n", key_dn);
+					Key * dn_key = ksLookupByName (returned, key_dn, 0);
+					if (last_dn != NULL)
+					{
+						elektraFree (last_dn);
+						if (fputs ("\n", lfp->fp) == EOF)
+						{
+							ELEKTRA_SET_VALIDATION_SYNTACTIC_ERRORF (
+								parentKey, "Could not write to position %ld in file %s", ftell (lfp->fp),
+								filename);
+							ldif_close (lfp);
+							return ELEKTRA_PLUGIN_STATUS_ERROR;
+						}
+					}
+					last_dn = key_dn;
+					const char * dn = keyString (dn_key);
+					char * data = ldif_put_wrap (LDIF_PUT_VALUE, "dn", dn, strlen (dn), LDIF_LINE_WIDTH);
+					if (fputs (data, lfp->fp) == EOF)
+					{
+						ELEKTRA_SET_VALIDATION_SYNTACTIC_ERRORF (
+							parentKey, "Could not write to position %ld in file %s", ftell (lfp->fp), filename);
+						ber_memfree (data);
+						ldif_close (lfp);
+						return ELEKTRA_PLUGIN_STATUS_ERROR;
+					}
+					ber_memfree (data);
+				}
+				else
+				{
+					elektraFree (key_dn);
+				}
+			}
+
+			const char * dn = keyString (cur);
+			ELEKTRA_LOG_DEBUG ("processing name: '%s', curr: '%s':'%s'\n", elektraKeyGetRelativeName (cur, parentKey), dn,
+					   keyBaseName (cur));
+			char * data = ldif_put_wrap (LDIF_PUT_VALUE, type, dn, strlen (dn), LDIF_LINE_WIDTH);
+
+			if (fputs (data, lfp->fp) == EOF)
+			{
+				ELEKTRA_SET_VALIDATION_SYNTACTIC_ERRORF (parentKey, "Could not write to position %ld in file %s",
+									 ftell (lfp->fp), filename);
+				ber_memfree (data);
+				ldif_close (lfp);
+				return ELEKTRA_PLUGIN_STATUS_ERROR;
+			}
+
+			ber_memfree (data);
+		}
+		else
+		{
+			ELEKTRA_LOG_DEBUG ("not processing name: '%s', curr: '%s':'%s'\n", elektraKeyGetRelativeName (cur, parentKey),
```

Comment:
This is because ldif does not allow any `=` in attribute names. If such case happens e.g. someone stores a value to a key with `=` in the last part of it, I see no other possibility than ignoring it. But you are right, a warning would be better in this case.